### ISSUE TITLE
UI - Add enable tls field in the logging target syslog

### ISSUE THREAD

**Issue by aiwantaozi** [rancher/rancher#17829]
Add enable tls field in the logging target syslog, this is for check whether user needs to enable tls for syslog tcp connect.

**Comment by n313893254** [comment]
UI for this https://github.com/rancher/rancher/issues/15380

**Comment by cjellick** [comment]
backend is in

**Comment by davidnuzik** [comment]
Version: Master 2.2 (2/13/19)
Confirmed feature component. An option to Use TLS was added to Syslog target option under Logging. I validated the UI to ensure all fields are working appropriately. I confirmed that we can only enable TLS when we set TCP for the endpoint. I validated that the HTTP request looks appropriate and that the API looks good.

Screenshot of this area in the UI (Syslog target)
![image](https://user-images.githubusercontent.com/45179589/52727443-78bde980-2f72-11e9-9ce7-6aac6e0ac0e0.png)

Req payload snippet:
```
syslogConfig: {certificate: "cacertPEM", clientCert: "ClientCert", clientKey: "ClientPrivateKey", enableTls: true,…}
certificate: "cacertPEM"
clientCert: "ClientCert"
clientKey: "ClientPrivateKey"
enableTls: true
endpoint: "159.65.64.22:8443"
program: "myprogram"
protocol: "tcp"
severity: "debug"
sslVerify: true
type: "/v3/schemas/syslogConfig"
```

Server response is good.
https://<server>/v3/clusterloggings (API) looks good.

Tested with various options. Ensure when use Test button fields are validated and you will get appropriate errors when you have invalid certs, etc.
